### PR TITLE
API with hooks

### PR DESCRIPTION
**Pull request by roosnic1**
but for now we call them events 😄

### CODE REVIEW

**Review by bishoymelek — DISMISSED** [review]

**Review by penk85 — APPROVED** [review]

File: packages/api/src/server.ts [hunk 1]
```diff
@@ -17,6 +20,40 @@ export class WepublishServer {
   constructor(opts: WepublishServerOpts) {
     const app = express()
 
+    const {dbAdapter} = opts
+
+    methodsToProxy.forEach(mtp => {
+      if (mtp.key in dbAdapter) {
+        mtp.methods.forEach(method => {
+          const methodName = `${method}${capitalizeFirstLetter(mtp.key)}`
+          // @ts-ignore
```
Comment (0815pascal) [inline packages/api/src/server.ts:29]: Did you leave those `@ts-ignore` on purpose here?

**Review by 0815pascal — DISMISSED** [review]
All good, only one question concerning `@ts-ignore`

File: packages/api/src/server.ts (hunk 1 above)
Comment (roosnic1) [reply packages/api/src/server.ts:29]: Yes. If i leave the out typescript complains and at the moment i have no idea how to type this.

Comment (faebebin) [reply packages/api/src/server.ts:29]: Think would have a fix there, will push later.

**Review by faebebin — CHANGES_REQUESTED** [review]

**Review by faebebin — DISMISSED** [review]
Looks good. And all checks pass locally too.

**Review by penk85 — APPROVED** [review]

**Review by 0815pascal — APPROVED** [review]
Looks good

**Pull request merged**